state/azure/tablestorage/tablestorage.go

```diff
@@ -311,7 +364,7 @@ func (r *StateStore) unmarshal(row *storage.Entity) ([]byte, *string, error) {
 	}
 
 	// use native ETag
-	etag := row.OdataEtag
+	etag := myEntity.ETag
 
 	return []byte(sv), ptr.String(etag), nil
```

Comment:
The Azure SDK have their own `to.Ptr(etag)` if you want to avoid using a different package

Reply:
I don't need the Ptr stuff in the first place. Removed, thanks